core/lib/zksync_core/src/api_server/web3/mod.rs

```diff
@@ -468,6 +459,16 @@ impl FullApiParams {
         local_addr_sender: oneshot::Sender<SocketAddr>,
         health_updater: HealthUpdater,
     ) -> anyhow::Result<()> {
+        // We don't want to start the server until at least one L1 batch is stored in Postgres.
+        let earliest_l1_batch_number =
+            wait_for_l1_batch(&self.pool, self.polling_interval, &mut stop_receiver)
+                .await
+                .context("error while waiting for L1 batch in Postgres")?;
```

Comment:
can we print a log line that we are waiting for it and when it's there? Just to make it easier to debug if we see api server not starting/not accepting requests. If we do this we can also output the block number